### ISSUE TITLE
在fluid中对两个loss分别计算梯度，回传更新参数

### ISSUE THREAD

**Issue by Emma-Ding** [PaddlePaddle/Paddle#19055]
![f2271c66968e9ba5bbf7eff99](https://user-images.githubusercontent.com/22070158/62601500-15f74a80-b924-11e9-9a78-e832ed384d1d.png)

`
def train(pserver_endpoints, training_role, current_endpoint, trainer_num, trainer_id, pserver_id, batch_size, use_parallel_executor=False, use_cuda=False, is_sparse=True, is_local=True):

    #data_list, predict, auc_var, cur_auc, auc_var_cos, cur_auc_cos, avg_cost, label, py_reader = net.model()
    data_list, predict_cos,  auc_var, cur_auc, avg_cost,avg_shantou, label, py_reader = net.model()
    sgd_optimizer = fluid.optimizer.Adam(learning_rate=0.0002)
    optimize_ops, params_grads = sgd_optimizer.minimize(avg_cost)
    optimize_ops_a, params_grads_a = sgd_optimizer.minimize(avg_cost_shantou)

    place = fluid.CUDAPlace(0) if use_cuda else fluid.CPUPlace()
    exe = fluid.Executor(place)

    data_name_list = [data.name for data in data_list]

    feeder = fluid.DataFeeder(feed_list=data_list, place=place)

    ctr = ctrdata.CTRData()

    with open('main.proto', 'w') as f:
        f.write(str(fluid.default_main_program()))

    def train_loop(main_program, trainer_id=None):
        """train loop
        """
        # prepare data`
如图配置，并不行，求问要怎么配置

**Comment by Xreki** [comment]
models中有一些任务，如图像检测、分类，里面都用到了多个loss，可以参考一下，一般用法是**将多个loss加到一起再求minimize**。例如：

https://github.com/PaddlePaddle/models/blob/6e12f8a1e734ad2f01169687fe60e95ba60652c0/PaddleCV/image_classification/train.py#L247

**Comment by Emma-Ding** [comment]
没有办法分开求梯度吗，我现在遇到的问题就是需要分开求梯度比较好

**Comment by dkerjw1** [comment]
> models中有一些任务，如图像检测、分类，里面都用到了多个loss，可以参考一下，一般用法是**将多个loss加到一起再求minimize**。例如：
> 
> https://github.com/PaddlePaddle/models/blob/6e12f8a1e734ad2f01169687fe60e95ba60652c0/PaddleCV/image_classification/train.py#L247

@Xreki  有同样的问题，求解，如何分别求梯度

**Comment by Xreki** [comment]
@Emma-Ding @QingyueMeng 跟同事请教了下，得到一些结论：
1）Paddle不支持一个program里面多个minimize。别的框架应该也不支持吧？
2）对于每个loss分别求梯度再回传，和先对多个loss求和，再统一求梯度回传，效果应该是一样的。

**Comment by zhiAung** [comment]
> @Emma-Ding @QingyueMeng 跟同事请教了下，得到一些结论：
> 1）Paddle不支持一个program里面多个minimize。别的框架应该也不支持吧？
> 2）对于每个loss分别求梯度再回传，和先对多个loss求和，再统一求梯度回传，效果应该是一样的。

您的意思是，把损失求和之后，在回传时，不同的分支只会使用损失和里面各自的损失进行优化，然后共享部分才会使用损失的总和进行优化吗？那这样的话，对不同的损失设置不同的权重是怎么影响网络的呢
